samples/java-protobuf-web-resources/src/main/java/com/example/FileServiceAction.java

```diff
@@ -0,0 +1,80 @@
+package com.example;
+
+import com.google.api.HttpBody;
+import com.google.protobuf.ByteString;
+import kalix.javasdk.Metadata;
+import kalix.javasdk.action.ActionCreationContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+// This class was initially generated based on the .proto definition by Kalix tooling.
+// This is the implementation for the Action Service described in your com.example/web_resources.proto file.
+//
+// As long as this file exists it will not be overwritten: you can maintain it yourself,
+// or delete it so it is regenerated as needed.
+
+public class FileServiceAction extends AbstractFileServiceAction {
+
+  private String defaultContentType = "text/html";
+  private Map<String, String> contentTypes = new HashMap();
+  private static final Logger LOG = LoggerFactory.getLogger(FileServiceAction.class);
+  private String baseDir = "src/main/resources/web";
+
+  public FileServiceAction(ActionCreationContext creationContext) {
+    contentTypes.put(".js","text/javascript");
+    contentTypes.put(".css","text/css");
+    contentTypes.put(".png","image/png");
+  }
+
+  private String getContentTypeByFile(String fileName) {
+    for (Map.Entry<String, String> entry : contentTypes.entrySet()) {
+      if(fileName.endsWith(entry.getKey())){
+        return entry.getValue();
+      }
+    }
+    return defaultContentType;
+  }
+
+  private  Effect<HttpBody> loadFile(String dir, String file) {
+    String fullPath = baseDir+dir+"/"+file;
+    try {
+      // tag::200-ok[]
+      byte[] byteArray = Files.readAllBytes(Paths.get(fullPath));
+      String contentType = getContentTypeByFile(file);
+      HttpBody response = HttpBody.newBuilder()
+              .setContentType(contentType)
+              .setData(ByteString.copyFrom(byteArray))
+              .build();
+      LOG.info("Serving {} with {}", fullPath, contentType);
+      Metadata header = Metadata.EMPTY.add("Cache-Control", "no-cache");
+      return effects().reply(response, header);
```

Comment:
I think there is room for introducing a small utility for building HttpBody

Something like: `effects().replyHttpBody("/path/to/some/file")`

We would take care of all the bits for the user. Of course, they would also have the option to build the HttpBody the way they prefer, add the headers they want and so forth, tweak the content type, etc. 

But for the most common cases (javascript, CSS, images and HTML), we could probably take care of them.

Reply:
The drawback that I see is that the Java SDK (code-first) should not expose `HttpBody` as a type. This is only needed for gRPC.

Reply:
This utility you have in mind would be a feature-request for the main sdk, right?

Does that mean, this is currently not possible with the code-first api?

Reply:
I think that level of API would be moving too much into the direction of "this is how we expect users to do things", think of this as a neat niche/side possibility.

Reply:
> Does that mean, this is currently not possible with the code-first api?

@sebastian-alfers, in code-first we can return a simple String as body, but the content type is set to `application/json`. Most probably we will have to open it up if we want to support this.